holoviews/plotting/bokeh/plot.py

```diff
@@ -103,12 +103,20 @@ def root(self):
 
     @document.setter
     def document(self, doc):
+        if (doc and hasattr(doc, 'on_session_destroyed') and
+            self.root is self.handles.get('plot') and not isinstance(self, AdjointLayoutPlot)):
+            doc.on_session_destroyed(self._session_destroy)
+            if self._document:
+                self._document._session_destroyed_callbacks.pop(self._session_destroy, None)
+
         self._document = doc
         if self.subplots:
             for plot in self.subplots.values():
                 if plot is not None:
                     plot.document = doc
 
+    def _session_destroy(self, session_context):
```

Comment:
So the only reason this method exists is to drop `session_context` before calling `cleanup`?